src/analyzer/protocol/ssh/ssh-analyzer.pac

```diff
@@ -203,12 +204,12 @@ refine flow SSH_Flow += {
 		%{
 		if ( ssh_server_host_key )
 			{
-			unsigned char digest[ZEEK_MD5_DIGEST_LENGTH];
-			zeek::detail::internal_md5(${key}.data(), ${key}.length(), digest);
+			unsigned char digest[ZEEK_SHA_DIGEST_LENGTH];
+			zeek::detail::internal_sha1(${key}.data(), ${key}.length(), digest);
 
 			zeek::BifEvent::enqueue_ssh_server_host_key(connection()->zeek_analyzer(),
 				connection()->zeek_analyzer()->Conn(),
-				zeek::make_intrusive<zeek::StringVal>(fingerprint_md5(digest)));
+				zeek::make_intrusive<zeek::StringVal>(fingerprint_sha1(digest)));
```

Comment:
Yes, this could definitely break correlation with data from prior versions. In its current form this PR captures more a _minimal_ changeset. Johanna, Christian and me had a call to start a discussion around that and the main points regarding this issue probably are

- If this code was written today we probably wouldn't use MD5 but another hash.
- Under certain circumstances breaking the data format in incompatible ways might be worthwhile.
- It is not clear at all whether one would need to certify Zeek for FIPS since it doesn't handle data at rest (which FIPS is concerned with), so we unsure whether #2505 is worth fixing in general

If this PR lands at all let's we definitely need to more carefully look at what APIs we break and whether we need to provide escape hatches.